### PR TITLE
fix: list & remove serving & failed flows

### PR DESCRIPTION
**Pull request by npitsillos**
**Goal**

Fixes https://github.com/jina-ai/jina-operator/issues/279

Changes the behaviour of:

* `jc list` where if no value is passed all flows that are in `Serving` or `Failed` phase are listed
* `jc remove` where if `all` is passed all flows that are in `Serving` or `Failed` phase are deleted

- [x] Run [Integration tests GHA](https://github.com/jina-ai/jcloud/actions/workflows/integration-tests.yml) manually & comment the link. https://github.com/jina-ai/jcloud/actions/runs/4073939574

`jc list`
image

`jc remove`
image

Flows are removed
image

@jina-ai/team-wolf

### CODE REVIEW

File: jcloud/api.py [hunk 1]
```diff
@@ -215,7 +221,9 @@ async def remove(args):
                 print('[cyan]No worries. Exiting...[/cyan]')
                 return
 
-        _raw_list = await _list_by_phase(Phase.Serving.value, name='')
+        _raw_list = await _list_by_phase(
```
Comment (subbuv26) [inline jcloud/api.py:224]: Update the comment `# Case 3: remove all ALIVE flows.` to `# Case 3: remove all ALIVE and DELETED flows.`

**Review by subbuv26 — CHANGES_REQUESTED** [review]

File: jcloud/api.py (hunk 1 above)
Comment (npitsillos) [reply jcloud/api.py:224]: will do

File: jcloud/api.py [hunk 2]
```diff
@@ -204,18 +205,18 @@ async def remove(args):
 
         flow_id_list = args.flows
 
-    # Case 3: remove all ALIVE flows.
+    # Case 3: remove all ALIVE and FAILED flows.
     else:
         if 'JCLOUD_NO_INTERACTIVE' not in os.environ:
             confirm_deleting_all = Confirm.ask(
-                f'[red]Are you sure you want to delete ALL the ALIVE flows that belong to you?[/red]',
+                f'[red]Are you sure you want to delete ALL the ALIVE and FAILED flows that belong to you?[/red]',
```
Comment (zac-li) [inline jcloud/api.py:212]: Serving and Failed

File: jcloud/api.py [hunk 3]
```diff
@@ -204,18 +205,18 @@ async def remove(args):
 
         flow_id_list = args.flows
 
-    # Case 3: remove all ALIVE flows.
+    # Case 3: remove all ALIVE and FAILED flows.
```
Comment (zac-li) [inline jcloud/api.py:208]: Serving and Failed

**Review by zac-li — CHANGES_REQUESTED** [review]
lgtm otherwise

File: jcloud/api.py [hunk 4]
```diff
@@ -152,11 +151,13 @@ async def _list_by_phase(phase: str, name: str):
     )
 
     console = Console(highlighter=CustomHighlighter())
-    msg = f'[bold]Fetching [green]{phase}[/green] flows'
+    if not phase:
+        phase = f'{Phase.Serving.value},{Phase.Failed.value}'
```
Comment (deepankarm) [inline jcloud/api.py:155]:
```suggestion
    if phase is None:
        phase = ','.join([str(Phase.Serving.value), str(Phase.Failed.value)])
```

File: jcloud/api.py [hunk 5]
```diff
@@ -204,18 +205,18 @@ async def remove(args):
...
-                f'[red]Are you sure you want to delete ALL the ALIVE flows that belong to you?[/red]',
+                f'[red]Are you sure you want to delete ALL the ALIVE and FAILED flows that belong to you?[/red]',
                 default=True,
             )
             if not confirm_deleting_all:
                 print('[cyan]No worries. Exiting...[/cyan]')
                 return
 
-        _raw_list = await _list_by_phase(Phase.Serving.value, name='')
+        _raw_list = await _list_by_phase(phase=None, name='')
```
Comment (deepankarm) [inline jcloud/api.py:219]: Better send the phases `Serving,Failed` here, instead of assuming default values in `_list_by_phase` if it is None.

**Review by deepankarm — CHANGES_REQUESTED** [review]

**Review by subbuv26 — APPROVED** [review]
LGTM

**Review by deepankarm — APPROVED** [review]

**Pull request merged**